src/overlays/actors/ovl_Obj_Lightblock/z_obj_lightblock.c

```diff
@@ -22,46 +27,178 @@ const ActorInit Obj_Lightblock_InitVars = {
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3AC8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3ADC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3B8C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3BA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3C18.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/func_80AF3C34.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lightblock/ObjLightblock_Draw.s")
+extern ColliderCylinderInit sCylinderInit;
+extern InitChainEntry sInitChain[];
+
+extern CollisionHeader D_06000B80;
+extern Gfx D_801AEF88[];
+extern Gfx D_06000178[];
+extern Gfx D_801AEFA0[];
+
+void func_80AF3910(ObjLightblock* this, GlobalContext* globalCtx) {
+    LightblockTypeVars* typeVars = &sLightblockTypeVars[LIGHTBLOCK_TYPE(&this->dyna.actor)];
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0x48, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
+                this->dyna.actor.world.pos.z, 0, 0, 0, typeVars->params);
+}
+
+void ObjLightblock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLightblock* this = THIS;
+    LightblockTypeVars* typeVars = &sLightblockTypeVars[LIGHTBLOCK_TYPE(&this->dyna.actor)];
+    s32 pad;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    Actor_SetScale(&this->dyna.actor, typeVars->scale);
+    BcCheck3_BgActorInit(&this->dyna, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    if (Flags_GetSwitch(globalCtx, LIGHTBLOCK_DESTROYED(&this->dyna.actor))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000B80);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+        Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+        this->collider.dim.radius = typeVars->radius;
+        this->collider.dim.height = typeVars->height;
+        this->collider.dim.yShift = typeVars->yShift;
+        this->opacity = 255;
+        func_80AF3AC8(this);
+    }
+}
+
+void ObjLightblock_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLightblock* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80AF3AC8(ObjLightblock* this) {
+    this->actionFunc = func_80AF3ADC;
+}
+
+void func_80AF3ADC(ObjLightblock* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        // light arrows
+        if (this->collider.info.acHitInfo->toucher.dmgFlags & 1 << 13) {
```

Comment:
```suggestion
        if (this->collider.info.acHitInfo->toucher.dmgFlags & (1 << 13)) {
```

Reply:
Hopefully we can get some damage flags to replace these soon.